packages/block-editor/src/components/block-media-update-progress/test/index.native.js

```diff
@@ -0,0 +1,501 @@
+	};
+	const subscribeMediaUpload = ( callback ) => {
+		this.uploadCallBack = callback;
+	};
+	const subscribeMediaSave = ( callback ) => {
+		this.saveCallBack = callback;
+	};
+	const mediaSources = {
+		deviceCamera: 'DEVICE_CAMERA',
+		deviceLibrary: 'DEVICE_MEDIA_LIBRARY',
+		siteMediaLibrary: 'SITE_MEDIA_LIBRARY',
+	};
+	return {
+		subscribeMediaUpload,
+		subscribeMediaSave,
+		sendMediaUpload: callUploadCallback,
+		sendMediaSave: callSaveCallback,
+		mediaSources,
+	};
+} );
+
+// const MEDIA_ID_REMOTE = 123;
+const MEDIA_ID_LOCAL = 2;
+const MEDIA_ID_TEMP = 'tempid-0-1';
+
+const tempMediaFiles = [
+	{
+		alt: '',
+		caption: '',
+		id: 'tempid-0-1',
+		link: '',
+		mime: 'image/jpeg',
+		type: 'image',
+		url: '',
+	},
+	{
+		alt: '',
+		caption: '',
+		id: 'tempid-0-2',
+		link: '',
+		mime: 'image/jpeg',
+		type: 'image',
+		url: '',
+	},
+];
+
+const localMediaFiles = [
+	{
+		alt: '',
+		caption: '',
+		id: '2',
+		link: '',
+		mime: 'image/jpeg',
+		type: 'image',
+		url: '',
+	},
+	{
+		alt: '',
+		caption: '',
+		id: '3',
+		link: '',
+		mime: 'image/jpeg',
+		type: 'image',
+		url: '',
+	},
+];
+
+describe( 'BlockMediaUpdateProgress component', () => {
+	it( 'renders without crashing', () => {
+		const wrapper = shallow(
+			<BlockMediaUpdateProgress renderContent={ () => {} } />
+		);
+		expect( wrapper ).toBeTruthy();
+	} );
+
+	it( 'upload: listens media upload progress for local file', () => {
+		const progress = 10;
+		const payload = {
+			state: MEDIA_UPLOAD_STATE_UPLOADING,
+			mediaId: MEDIA_ID_LOCAL,
+			progress,
+		};
+
+		const onUpdateMediaUploadProgress = jest.fn();
+
+		const wrapper = shallow(
+			<BlockMediaUpdateProgress
+				onUpdateMediaUploadProgress={ onUpdateMediaUploadProgress }
+				mediaFiles={ localMediaFiles }
+				renderContent={ () => {} }
+			/>
+		);
+
+		sendMediaUpload( payload );
+
+		expect( wrapper.instance().state.progress ).toEqual( progress );
+		expect( wrapper.instance().state.isUploadInProgress ).toEqual( true );
+		expect( wrapper.instance().state.isUploadFailed ).toEqual( false );
+		expect( onUpdateMediaUploadProgress ).toHaveBeenCalledTimes( 1 );
+		expect( onUpdateMediaUploadProgress ).toHaveBeenCalledWith( payload );
+	} );
+
+	// UPLOAD tests
+	it( 'upload does not get affected by unrelated media uploads', () => {
+		const payload = {
+			state: MEDIA_UPLOAD_STATE_UPLOADING,
+			mediaId: 432, // id not belonging to assigned mediaFiles collection in test
+			progress: 20,
+		};
+		const onUpdateMediaUploadProgress = jest.fn();
+		const wrapper = shallow(
+			<BlockMediaUpdateProgress
+				onUpdateMediaUploadProgress={ onUpdateMediaUploadProgress }
+				mediaFiles={ localMediaFiles }
+				renderContent={ () => {} }
+			/>
+		);
+
+		sendMediaUpload( payload );
+
+		expect( wrapper.instance().state.progress ).toEqual( 0 );
+		expect( onUpdateMediaUploadProgress ).toHaveBeenCalledTimes( 0 );
+	} );
+
+	it( 'upload: listens media upload success', () => {
```

Comment:
@mzorz can we make these test descriptions a sentence/phrase without the `upload:` ? Let me know if it's a paradigm I am missing 🙏

Reply:
Could we also make the purpose of this test a bit more descriptive in what it does? I am seeing the description and logic at the `expect` calls and they don't correlate easily for me. Let me know 🙏

Reply:
>  can we make these test descriptions a sentence/phrase without the upload: ? Let me know if it's a paradigm I am missing 🙏

No paradigm missed here 👍 , it made sense since there's a very similar set of tests testing for the same conditions on two different operations: upload and save, that I prefixed each test with that. So I think it's clearer given the nature of the component which can handle progress updates on these two operations. I think it's fine to keep it, unless you feel strongly against that.

> Could we also make the purpose of this test a bit more descriptive in what it does? I am seeing the description and logic at the expect calls and they don't correlate easily for me. Let me know 

the name of the test is `listens to media upload sucess` -  the assertions are about making sure the `onFinishMediaUploadWithSuccess` method gets called once and with the success test payload. Do you have anything in mind that better describes the purpose of the test? Let me know 🙏

Reply:
> No paradigm missed here 👍 , it made sense since there's a very similar set of tests testing for the same conditions on two different operations: upload and save, that I prefixed each test with that. So I think it's clearer given the nature of the component which can handle progress updates on these two operations. I think it's fine to keep it, unless you feel strongly against that.

Understood! Yes, it's fine to keep it. 🤝 

> the name of the test is listens to media upload sucess - the assertions are about making sure the onFinishMediaUploadWithSuccess method gets called once and with the success test payload. Do you have anything in mind that better describes the purpose of the test? Let me know 🙏

Thanks for the explanation. I sometimes put the name of the method/target in the name of the method. I came up with this below
`onFinishMediaUploadWithSuccess is called when a success payload is received`

That's what I came up with :) If it's okay with you we could use it or keep the one that's there 🤝

Reply:
Nice! I changed that one and the rest to comply with the same naming pattern in 1f57e1d7e6

Reply:
Sounds good!